vumi/persist/txriak_manager.py

```diff
@@ -155,31 +159,60 @@ def set_user_metadata(self, usermeta):
         self._riak_obj.usermeta = usermeta
 
     def get_bucket(self):
-        return VumiTxRiakBucket(self._riak_obj.bucket)
+        return VumiTxRiakBucket(self._riak_obj.bucket, self._manager)
 
     # Methods that touch the network.
 
     def store(self):
-        d = deferToThread(self._riak_obj.store)
-        d.addCallback(type(self))
+        d = self._manager.deferToThread(self._riak_obj.store)
+        d.addCallback(type(self), self._manager)
         return d
 
     def reload(self):
-        d = deferToThread(self._riak_obj.reload)
-        d.addCallback(type(self))
+        d = self._manager.deferToThread(self._riak_obj.reload)
+        d.addCallback(type(self), self._manager)
         return d
 
     def delete(self):
-        d = deferToThread(self._riak_obj.delete)
-        d.addCallback(type(self))
+        d = self._manager.deferToThread(self._riak_obj.delete)
+        d.addCallback(type(self), self._manager)
         return d
 
 
+def _daemon_threadpool(*args, **kw):
+    """
+    Create a threadpool that sets its threads to daemon mode so they don't
+    block process exit if they aren't properly cleaned up.
```

Comment:
I should note that I discovered this stuff because my test run processes never terminated and I had to `kill -9` them. Consider that behaviour in production when a worker process fails or is restarted.